test/unit/assorted/deprecate_warning.test.js

```diff
@@ -195,25 +192,12 @@ describe('Deprecation Warnings', function () {
     });
   });
 
+  // TODO(NODE-4817): Do we need these tests?
```

Comment:
Are these tests necessary?

Reply:
Actually this entire test can be removed as well as `deprecateOptions` from utils - it's not used anywhere in the codebase that I can see and there is no test that uses it outside of this one. There may be even more clean-up as well. `ClassWithoutLogger` and `ClassWithUndefinedLogger` can also be removed from test/tools/utils. Nice check.